lib/vanagon/platform/windows.rb

```diff
@@ -0,0 +1,53 @@
+class Vanagon
+  class Platform
+    class Windows < Vanagon::Platform
+      # The specific bits used to generate a windows package for a given project
+      #
+      # @param project [Vanagon::Project] project to build a windows package of
+      # @return [Array] list of commands required to build a windows package for the given project from a tarball
+      def generate_package(project)
+        target_dir = project.repo ? output_dir(project.repo) : output_dir
+      end
+
+      # Method to generate the files required to build a windows package for the project
+      #
+      # @param workdir [String] working directory to stage the evaluated templates in
+      # @param name [String] name of the project
+      # @param binding [Binding] binding to use in evaluating the packaging templates
+      def generate_packaging_artifacts(workdir, name, binding)
+        win_dir = File.join(workdir, "windows")
+        FileUtils.mkdir_p(win_dir)
+
+        # Lots of templates here
+        [""].each do |win_file|
+          erb_file(File.join(VANAGON_ROOT, "templates/win/#{win_file}.erb"), File.join(win_dir, win_file), false, { :binding => binding })
+        end
+      end
+
+      # Method to derive the package name for the project
+      #
+      # @param project [Vanagon::Project] project to name
+      # @return [String] name of the windows package for this project
+      def package_name(project)
+        "#{project.name}-#{project.version}-#{project.release}-#{@architecture}.msi"
```

Comment:
Are noarch projects a thing in windows or will there always be an arch specified?